lib/tasks/seed.rake

```diff
@@ -0,0 +1,139 @@
+require 'factory_girl'
+
+namespace :seed do
+  desc 'Generate various forgeries. This should not be used in production'
+  task :forgeries => %w[forgeries:all]
+
+  namespace :forgeries do
+    task :load_factories => :environment do
+      if Rails.env.prodution?
+        STDERR.puts "Refusing to seed data in a production environment"
+        exit 1
+      end
+
+      include FactoryGirl::Syntax::Methods
+      FactoryGirl.find_definitions
+    end
+
+    task :domains => :load_factories do
+      names = ['example.com', 'example.org', 'example.net']
+
+      if SETTINGS[:organizations_enabled]
+        Organization.all.each do |organization|
+          Organization.current = organization
+          generate(Domain, get_desired(5)) do |count, generator|
+            create(:domain,
+                   :name => build_domain(names, generator),
+                   :organizations => [organization])
+          end
+        end
+      else
+        generate(Domain, get_desired(50)) do |count, generator|
+          create(:domain,
+                 :name => build_domain(names, generator))
+        end
+      end
+    end
+
+    task :hosts => :load_factories do
+      domains = Domain.all
+      if SETTINGS[:organizations_enabled]
+        organizations = Organization.all
+      end
+      operatingsystems = Operatingsystem.all
+      owner = User.anonymous_admin
+
+      generate(Host::Managed, get_desired(100)) do |count, generator|
+        os = operatingsystems.sample
+
+        if SETTINGS[:organizations_enabled]
+          Organization.current = organizations.sample
+          domains = Domain.all
+        end
+
+        create(:host,
+               :hostname => generator.next_random_name,
+               :domain => domains.sample,
+               :operatingsystem => os,
+               :architecture => os.architectures.sample,
+               :organization => Organization.current,
+               :owner => owner)
+      end
+    end
+
+    task :operatingsystems => :load_factories do
+      build_operatingsystems.each do |os|
+        unless Operatingsystem.find_by(:name => os.name, :major => os.major, :minor => os.minor)
+          os.save!
+          puts "Created #{os}"
+        end
+      end
+    end
+
+    task :organizations => :load_factories do
+      if SETTINGS[:organizations_enabled]
+        generate(Organization, get_desired(10)) do |count, generator|
+          create(:organization, :name => generator.next_random_name)
+        end
+      end
+    end
+
+    task :all => [:operatingsystems, :organizations, :domains, :hosts] do
+    end
+
+    def get_desired(default)
+      (ENV['COUNT'] || default).to_i
```

Comment:
For example if I want just 50 hosts (instead of the default 100), setting `COUNT` to 50 would lead to 50 orgs being created. Could something be done about this?

Reply:
I considered that and we could probably get some context so you can use `ORGANIZATION_COUNT` and `DOMAIN_COUNT` by moving it into the `generate` function.